edxval/tests/test_api.py

```diff
@@ -436,3 +641,75 @@ def test_get_info_queries_for_only_video(self):
         with self.assertNumQueries(4):
             api.get_video_info(constants.VIDEO_DICT_ZEBRA.get("edx_video_id"))
 
+
+class TestCopyCourse(TestCase):
+    """Tests rerun course"""
+
+    def setUp(self):
+        """
+        Creates a course for testing
+
+        Creates two videos, each with two encodings.
+        """
+        self.course_id = 'test-course'
+        # 1st video
+        video = Video.objects.create(**constants.VIDEO_DICT_FISH)
+        CourseVideo.objects.create(video=video, course_id=self.course_id)
+        # 2nd video
+        video = Video.objects.create(**constants.VIDEO_DICT_STAR)
+        CourseVideo.objects.create(video=video, course_id=self.course_id)
+
+        self.course_id2 = "test-course2"
+        # 3rd video different course
+        video = Video.objects.create(**constants.VIDEO_DICT_TREE)
+        CourseVideo.objects.create(video=video, course_id=self.course_id2)
+
+    def test_successful_copy(self):
+        """Tests a successful rerun and 2nd rerun"""
+        api.copy_course_videos('test-course', 'course-rerun1')
+        original_videos = Video.objects.filter(courses__course_id='test-course')
+        copied_videos = Video.objects.filter(courses__course_id='course-rerun1')
+        other_course = Video.objects.filter(courses__course_id='test-course2')
+
+        self.assertEqual(len(original_videos), 2)
+        self.assertEqual(original_videos[0].edx_video_id,
+                         constants.VIDEO_DICT_FISH['edx_video_id'])
+        self.assertEqual(original_videos[1].edx_video_id,
+                         constants.VIDEO_DICT_STAR['edx_video_id'])
+        self.assertTrue(set(original_videos) == set(copied_videos))
+
+        self.assertEqual(len(other_course), 1)
+        self.assertEqual(other_course[0].edx_video_id,
+                         constants.VIDEO_DICT_TREE['edx_video_id'])
+
+    def test_same_course_ids(self):
+        """Tests when the new course id name is the same as the original"""
+        original_videos = Video.objects.filter(courses__course_id='test-course')
+        api.copy_course_videos('test-course', 'test-course')
+        copied_videos = Video.objects.filter(courses__course_id='test-course')
+        self.assertEqual(len(original_videos), 2)
+        self.assertEqual(original_videos[0].edx_video_id,
+                         constants.VIDEO_DICT_FISH['edx_video_id'])
+        self.assertEqual(original_videos[1].edx_video_id,
+                         constants.VIDEO_DICT_STAR['edx_video_id'])
```

Comment:
I don't think these are really necessary; the only assertion that really matters is the one on the next line.